products/BellHybrid/services/db/databases/MeditationStatisticsTable.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "MeditationStatisticsTable.hpp"
+#include <time.h>
+#include <chrono>
+#include <date/date.h>
+
+namespace
+{
+    using namespace db::meditation_stats;
+    constexpr auto tableName = "meditation_stats";
+
+    /// We could use the SQLite function "datetime(timestamp,'unixepoch') to generate the correct time string.
+    /// Unfortunately, SQLite v3.38.1 has some problems converting UNIX timestamps. That's why we decided we would store
+    /// timestamps using string format, as it doesn't cause any issues.
+
+    std::string prepare_timestamp(time_t unix_timestamp)
+    {
+        char time_buf[64]{};
+        struct tm ts
+        {};
+        ts = *gmtime(&unix_timestamp);
+        strftime(time_buf, sizeof(time_buf), "%Y-%m-%d %H:%M:%S", &ts);
+
+        return time_buf;
+    }
+
+    time_t to_unixepoch_timestamp(std::string_view str)
+    {
+        std::istringstream in{str.data()};
+        date::sys_seconds tp;
+        in >> date::parse("%Y-%m-%d %H:%M:%S", tp);
+        return tp.time_since_epoch().count();
+    }
+
+    std::vector<TableRow> get_by_x(Database *db, const std::string_view modifier, const std::uint32_t x)
+    {
+        if (x == 0) {
+            return {};
+        }
+
```

Comment:
Did you consider checking that `db` is not `nullptr`?

Reply:
It's just an internal helper function. It assumes that the `Database` pointer is always valid. It is guaranteed as this pointer is passed to the constructor of `MeditationStatsTable`, and this function is only invoked from MeditationStatsTable's context.

Reply:
I get it, thanks for clarifying :)